src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -915,6 +933,19 @@ private void ParseHelper(string[] args)
                     _noInteractive = false;
                     ParametersUsed |= ParameterBitmap.Interactive;
                 }
+                else if (MatchSwitch(switchKey, "configurationfile", "configurationFile"))
+                {
+                    ++i;
+                    if (i >= args.Length)
+                    {
+                        SetCommandLineError(
+                            CommandLineParameterParserStrings.MissingConfigurationFileArgument);
+                        break;
+                    }
+
+                    _configurationFile = args[i];
```

Comment:
I don't feel this is necessary, at least for this initial check in.  The existing 'CreateFromSessionConfigurationFile' already performs these basic file checks and I feel it is not necessary to duplicate them.  I added extra validation for the .pssc contents because before it would just ignore unknown entries.  The session won't start if any error is thrown, and I feel that is a fast enough failure.  If this proves not to be the case, a check done at a sooner point can be created.

Reply:
After building this branch, assertions are thrown when running `pwsh.exe -configurationfile abc` in debug builds.
It's related to path resolution.

```
C:\Users\rocky>C:\arena\source\PowerShell\src\powershell-win-core\bin\Debug\net7.0\win7-x64\publish\pwsh.exe -configurationfile abc
PowerShell 7.3.0-preview.3-146-ga91c43bb5c46bedb8cf95c7140c536acd1d34c1d
Process terminated. Assertion failed.
Caller makes sure that 'path' is already resolved.
   at System.Management.Automation.Diagnostics.Assert(Boolean condition, String whyThisShouldNeverHappen, String detailMessage) in C:\arena\source\PowerShell\src\System.Management.Automation\utils\assert.cs:line 202
   at System.Management.Automation.Diagnostics.Assert(Boolean condition, String whyThisShouldNeverHappen) in C:\arena\source\PowerShell\src\System.Management.Automation\utils\assert.cs:line 134
   at System.Management.Automation.ExternalScriptInfo..ctor(String name, String path, ExecutionContext context) in C:\arena\source\PowerShell\src\System.Management.Automation\engine\ExternalScriptInfo.cs:line 50
   at System.Management.Automation.Remoting.DISCUtils.GetScriptInfoForFile(ExecutionContext context, String fileName, String& scriptName) in C:\arena\source\PowerShell\src\System.Management.Automation\engine\remoting\fanin\InitialSessionStateProvider.cs:line 1429
   at System.Management.Automation.Remoting.DISCPowerShellConfiguration..ctor(String configFile, Func`2 roleVerifier, Boolean validateFile) in C:\arena\source\PowerShell\src\System.Management.Automation\engine\remoting\fanin\InitialSessionStateProvider.cs:line 1756
   at System.Management.Automation.Runspaces.InitialSessionState.CreateFromSessionConfigurationFile(String path, Func`2 roleVerifier, Boolean validateFile) in C:\arena\source\PowerShell\src\System.Management.Automation\engine\InitialSessionState.cs:line 1354
   at Microsoft.PowerShell.ConsoleHost.DoCreateRunspace(RunspaceCreationEventArgs args) in C:\arena\source\PowerShell\src\Microsoft.PowerShell.ConsoleHost\host\msh\ConsoleHost.cs:line 1670
   at Microsoft.PowerShell.ConsoleHost.CreateRunspace(RunspaceCreationEventArgs runspaceCreationArgs) in C:\arena\source\PowerShell\src\Microsoft.PowerShell.ConsoleHost\host\msh\ConsoleHost.cs:line 1601
   at Microsoft.PowerShell.ConsoleHost.DoRunspaceLoop(String initialCommand, Boolean skipProfiles, Collection`1 initialCommandArgs, Boolean staMode, String configurationName, String configurationFilePath) in C:\arena\source\PowerShell\src\Microsoft.PowerShell.ConsoleHost\host\msh\ConsoleHost.cs:line 1521
   at Microsoft.PowerShell.ConsoleHost.Run(CommandLineParameterParser cpp, Boolean isPrestartWarned) in C:\arena\source\PowerShell\src\Microsoft.PowerShell.ConsoleHost\host\msh\ConsoleHost.cs:line 1495
   at Microsoft.PowerShell.ConsoleHost.Start(String bannerText, String helpText) in C:\arena\source\PowerShell\src\Microsoft.PowerShell.ConsoleHost\host\msh\ConsoleHost.cs:line 263
   at Microsoft.PowerShell.UnmanagedPSEntry.Start(String[] args, Int32 argc) in C:\arena\source\PowerShell\src\Microsoft.PowerShell.ConsoleHost\host\msh\ManagedEntrance.cs:line 99
   at Microsoft.PowerShell.ManagedPSEntry.Main(String[] args) in C:\arena\source\PowerShell\src\powershell\Program.cs:line 72
```

Reply:
I thought the public API provided checks and error messages, but clearly not.  I have added the checks.